examples/simple-inference/simple-inference.cpp

```diff
@@ -0,0 +1,401 @@
+// Defines sigaction on msys:
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
+#include "common.h"
+
+#include "console.h"
+#include "llama.h"
+#include "build-info.h"
+#include "grammar-parser.h"
+
+#include <atomic>
+#include <cassert>
+#include <cinttypes>
+#include <cmath>
+#include <cstdio>
+#include <cstring>
+#include <ctime>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#if defined (__unix__) || (defined (__APPLE__) && defined (__MACH__))
+#include <signal.h>
+#include <unistd.h>
+#elif defined (_WIN32)
+#define WIN32_LEAN_AND_MEAN
+#ifndef NOMINMAX
+#define NOMINMAX
```

Comment:
I don't understand why we use NOMINMAX everywhere instead of including system headers _before_ library headers, and using `#undef` if necessary.

Reply:
I don't really know anything about the Windows stuff, this part was just copied over from the `main` example. If there's a better way, I'm certainly open to applying it but I don't have a way to test that it doesn't break anything.